### PR TITLE
feat(balance): reference manuals no longer useless

### PR DESCRIPTION
**Pull request by chaosvolt**
### Required

- [x] I wrote the PR title in [conventional commit format](https://docs.cataclysmbn.org/en/contribute/changelog_guidelines/).
- [x] I ran the [code formatter](https://docs.cataclysmbn.org/en/contribute/contributing/#code-style).
- [x] I linked any relevant issues using [github keyword syntax](https://docs.cataclysmbn.org/en/contribute/contributing/#pull-request-notes) so it can be closed automatically.
- [x] I have [committed my changes to new branch that isn't `main`](https://docs.cataclysmbn.org/en/contribute/contributing/#make-your-changes) so it won't cause conflict when updating `main` branch later.

## Purpose of change

One minor thing on my todo list that had been bugging me for a while are the reference manual books we inherited from DDA, which clearly seem geared towards the proficiency system. They gave no skill training and were involved in no recipes. This finally makes them useful.

## Describe the solution

1. Added cooking training range of 6-10 to CRC-Merck Handbook, being a common advanced book range previously unused by cooking books. Also elevated intelligence threshold from 4 to 12 now that it's actually useful.
2. Added fabrication training range of 7-9 to Machinery's Handbook, seemed to be one of the least common skill ranges for fabrication books. Elevated intelligence threshold from 4 to 10.
3. Added first aid training range of 6-10 to Physicians' Desk Reference, another case of a fitting advanced-level skill range previously unused by first aid books. Elevated intelligence threshold from 4 to 12. Also removed the bits in the description that seem intent on justifying why it was actively useless as a first aid skillbook.
4. Added first aid training range of 2-5 to Merck Veterinary Manual. Not sure if it's too low, but it helps cover the common skill gap of hitting level 4 that can often serve as a wall to players, and improvising veterinary info for human first aid should likely cover a lower skill range regardless. Also bumped int requirement from 4 to 9. Lastly, removed the reference to animal cloning since that implies you'd it learn in this book, and that seems more likely to end up in an in-repo mod even in BN.
5. Updated several recipes to include these books as source of booklearns. CRC handbook mostly used for a mix of general chemistry recipes normally split between the other mid-high level chemistry books, machinery book got all the 3D printing recipes as well as a couple mid-high level fabrication and mechanics recipes, while the two first aid books gained several relevant medical and/or chemical recipes. One of the recipe inclusions in the veterinary manual is more of an easter egg/joke and could be removed if requested (find it first though :3).
6. Misc: fixed booklearns for carbon fiber bolts and arrows being different.

## Describe alternatives you've considered

Obsoleting them for being an obviously unfinished idea inherited from DDA (and obviously unfinished even in DDA, see below).

## Testing

Ported JSON files to playthrough release and load-tested. Decided I'd bully autofix with this one since I'm working while out and about on fairly shitty wifi.

## Additional context

I checked DDA and predictably, as of build `2024-07-07-1825`, the affected books not only don't level any skills, they don't even grant any proficiencies. That was literally what I assumed the bloody things were implemented for and they didn't even bother with that. They did add a tiny handful of recipe uses here and there but that's literally it (topical cocaine for the CRC book, steel lumps and a fanning mill for the machinery book, literally nothing for the first aid books).

Side note, remind me to follow up on this by sanity-checking and rebalancing the skill levels needed for 3D printing stuff, in addition to adding more recipes as fits.

There are probably also some autolearn-only recipes here and there that could be given reference books as a way to booklearn them early.

### CODE REVIEW

File: data/json/recipes/other/tools.json [hunk 1]
```diff
@@ -878,7 +879,7 @@
     "difficulty": 3,
     "time": "20 m",
     "autolearn": true,
-    "book_learn": [ [ "fun_survival", 2 ], [ "manual_fabrication", 2 ] ],
+    "book_learn": [ [ "fun_survival", 2 ], [ "manual_fabrication", 2 ], [ "reference_fabrication1", 3 ] ],
```
Comment (Lamandus) [inline data/json/recipes/other/tools.json:882]:
```suggestion
    "book_learn": [ [ "fun_survival", 2 ], [ "manual_fabrication", 2 ], [ "reference_fabrication1", 2 ] ],
```

Autolearn already is at lvl 3

File: data/json/recipes/other/materials.json [hunk 2]
```diff
@@ -500,7 +500,7 @@
     "charges": 10,
     "autolearn": false,
```
Comment (Lamandus) [inline data/json/recipes/other/materials.json:501]:
```suggestion
```
if we are already at it...

File: data/json/recipes/chem/other.json [hunk 3]
```diff
@@ -109,7 +109,8 @@
     "batch_time_factors": [ 83, 5 ],
     "autolearn": true,
     "charges": 10,
-    "book_learn": [ [ "recipe_arrows", 1 ], [ "textbook_survival", 1 ], [ "survival_book", 1 ] ],
+    "//": "Veterinary handbook teaching this recipe is more of an easter egg but is intentional",
+    "book_learn": [ [ "recipe_arrows", 1 ], [ "textbook_survival", 1 ], [ "survival_book", 1 ], [ "reference_firstaid2", 4 ] ],
```
Comment (Lamandus) [inline data/json/recipes/chem/other.json:113]:
```suggestion
    "book_learn": [ [ "recipe_arrows", 1 ], [ "textbook_survival", 1 ], [ "survival_book", 1 ], [ "reference_firstaid2", 2 ] ],
```

is lvl 4 even as an easter egg a bit too steep?

File: data/json/recipes/chem/fuel.json [hunk 4]
```diff
@@ -10,7 +10,7 @@
     "time": "30 m",
     "batch_time_factors": [ 83, 5 ],
     "autolearn": true,
-    "book_learn": [ [ "textbook_chemistry", 3 ], [ "brewing_cookbook", 3 ] ],
+    "book_learn": [ [ "textbook_chemistry", 3 ], [ "brewing_cookbook", 3 ], [ "reference_cooking", 4 ] ],
```
Comment (Lamandus) [inline data/json/recipes/chem/fuel.json:13]:
```suggestion
    "book_learn": [ [ "textbook_chemistry", 3 ], [ "brewing_cookbook", 3 ], [ "reference_cooking", 3 ] ],
```

autolearn is at lvl 4

**Review by Lamandus — CHANGES_REQUESTED** [review]

**Review by scarf005 — APPROVED** [review]

**Pull request merged**